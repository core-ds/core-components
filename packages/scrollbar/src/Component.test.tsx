import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { Scrollbar, ScrollbarProps } from './index';

const scrollableDataTestId = 'scrollable-node';
const scrollableNodeProps = { 'data-test-id': scrollableDataTestId };

const renderScrollbar = (props?: ScrollbarProps) => (
    <Scrollbar
        {...props}
        style={{ height: 200 }}
        scrollableNodeProps={{ ...props?.scrollableNodeProps, ...scrollableNodeProps }}
    >
        <div style={{ height: 500 }} />
    </Scrollbar>
);

describe('Scrollbar', () => {
    describe('Display tests', () => {
        it('should display correctly', () => {
            const { container } = render(renderScrollbar({ autoHide: false }));

            expect(container).toMatchSnapshot();
        });

        it('should visible x and y track', async () => {
            const { getByTestId } = render(renderScrollbar({ forceVisible: true }));

            const { style } = getByTestId(scrollableDataTestId);

            expect(style.overflowX).toBe('scroll');
            expect(style.overflowY).toBe('scroll');
        });
    });

    describe('Callbacks tests', () => {
        it('should call onScroll', async () => {
            const onScroll = jest.fn();

            const { getByTestId } = render(renderScrollbar({ scrollableNodeProps: { onScroll } }));

            fireEvent.scroll(getByTestId(scrollableDataTestId));

            expect(onScroll).toBeCalledTimes(1);
        });
    });

    describe('Render tests', () => {
        it('should unmount without errors', () => {
            const { unmount } = render(renderScrollbar());

            expect(unmount).not.toThrowError();
        });
    });
});
