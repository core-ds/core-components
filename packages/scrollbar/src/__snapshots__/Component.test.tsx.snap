// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollbar Display tests should display correctly 1`] = `
<div>
  <div
    class="component component"
    data-simplebar="init"
    style="height: 200px;"
  >
    <div
      class="wrapper"
    >
      <div
        class="heightAutoObserverWrapper"
      >
        <div
          class="heightAutoObserver"
        />
      </div>
      <div
        class="mask"
      >
        <div
          class="offset"
          style="right: 0px; bottom: 0px;"
        >
          <div
            aria-label="scrollable content"
            class="contentWrapper"
            data-test-id="scrollable-node"
            role="region"
            style="height: auto; overflow-x: hidden; overflow-y: hidden;"
            tabindex="0"
          >
            <div
              class="content"
            >
              <div
                style="height: 500px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="placeholder"
        style="width: 0px; height: 0px;"
      />
    </div>
    <div
      class="track horizontal"
      style="visibility: hidden;"
    >
      <div
        class="scrollbar visible"
        style="width: 0px; display: none;"
      />
    </div>
    <div
      class="track vertical"
      style="visibility: hidden;"
    >
      <div
        class="scrollbar visible"
        style="height: 0px; display: none;"
      />
    </div>
  </div>
</div>
`;
